pkg/fakerp/fakerp.go

```diff
@@ -118,7 +122,72 @@ func GetDeployer(log *logrus.Entry, cs *api.OpenShiftManagedCluster, testConfig
 			log.Warnf("deployment failed: %#v", err)
 			debugDeployerError(ctx, log, cs, err, testConfig)
 		}
-		return err
+
+		// This causes fails in our tests because deployer function is not versioned.
+		// TODO: Remove this when v7 is gone and PLS is default in the deployer function
+		if cs.Config.PluginVersion == "v9.0" &&
+			conf != nil {
```

Comment:
I'd like to understand this `conf != nil` check better.  In what circumstances are we allowing conf to be nil and why?

Reply:
use-case is - most of the geneva actions dont have config object. And they dont need to. It is specific to our fakeRP implementation

Reply:
Instead if checking this, check for `if len(pe) = 0 ` - create, else - ignore

Reply:
Put it in the middleware and get it from Context.